### DISCUSSION TITLE
Does jobs has any events defined by the core ?

### DISCUSSION THREAD

**Discussion by Blair2004** [laravel/framework#35686] [Q&A]
So I'm building an application, and I would like it to be multi-tenant (I build my own concept which works pretty well so far). To briefly explain the concept, every model extends a parent model which, when the getTable() method is called, returns a customized table name (in my case a sub-store table name). However, when I'm working with the job and let's say I dispatch a `HandleProcurementJob`, to which I'm passing a `Procurement` model, since it's running with `php artisan quueue:work`, i'm not able to execute a proper script before the job runs to change how the model will be unserialized (determining if the table name should include a store prefix or not). 

So my question is: is there a way to listen to even before a job runs on Laravel?  

*Laravel*: 8.x
*PHP* : 7.4

I'm pretty new to this forum, apologies if I'm doing it wrong.

**Comment by Wulfheart** [comment]
> So I'm building an application, and I would like it to be multi-tenant (I build my own concept which works pretty well so far). To briefly explain the concept, every model extends a parent model which, when the getTable() method is called, returns a customized table name (in my case a sub-store table name). 

Do I assume correctly that you are using scopes for this?

**Reply by Blair2004** [reply]
Not really, I'm using Laravel events and Tormjens/Eventy this way : 

1 - Laravel Events: This is used to dispatch an event when a route is being registered. Basically, that let me detect if the route has a specific structure for example "example.com/dashboard/store/1" and "example.com/dashboard" as one has a store slug, we'll then use...
2 - Tormjens/Eventy to mutate the model table name. So every time within that route a model is used for example "Procurement", the table name will include a prefix of the store being explored.

So, as I was planning to make jobs that perform tasks for sub stores, I wanted to be able to change (on certain conditions) the table name before a job runs. This so that a Model points to the right store.
